runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1233,7 +1233,7 @@ MM_ObjectAccessBarrier::copyObjectFieldsToFlattenedArrayElement(J9VMThread *vmTh
 	U_8 *elementAddress = (U_8*)indexableEffectiveAddress(vmThread, arrayRef, index, J9ARRAYCLASS_GET_STRIDE((J9Class *) arrayClazz));
 	IDATA elementOffset = (elementAddress - (U_8*)arrayRef);
 	J9Class *elementClazz = J9GC_J9OBJECT_CLAZZ_THREAD(srcObject, vmThread);
-	Assert_MM_true(J9_IS_J9CLASS_ALLOW_DEFAULT_VALUE(elementClazz));
+	Assert_MM_true(J9_IS_J9CLASS_VALUEBASED(elementClazz));
```

Comment:
You're looking for `J9_IS_J9CLASS_VALUETYPE` here. `J9_IS_J9CLASS_VALUEBASED ` refers to the ValueBased annotation.